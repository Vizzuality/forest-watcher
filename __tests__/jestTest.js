import 'react-native';
import appReducer, { setAppSynced, setOfflineMode, syncApp } from 'redux-modules/app';

// JEST TESTS
describe('Unit Tests', () => {
  it('test runs', () => {
    expect(true);
  });

  it('snapshot runs', () => {
    expect({ hello: true, number: 232 }).toMatchSnapshot();
  });
});

// ACTION TESTS
describe('Redux Actions - Unit Tests', () => {
  it('snapshot redux action test', () => {
    expect(setOfflineMode(true)).toMatchSnapshot();
    expect(setOfflineMode(false)).toMatchSnapshot();
  });

  it('snapshot redux action thunk test', () => {
    expect(syncApp()).toMatchSnapshot(); // thunked action tests don't work currently - no useful output
  });
});

// REDUCER & STATE TESTS
describe('Redux Reducers - Unit Tests', () => {
  it('snapshot redux reducer initial state test', () => {
    expect(appReducer(undefined, { type: 'NONE' })).toMatchSnapshot();
  });

  it('snapshot redux reducer action test', () => {
    expect(appReducer(undefined, setAppSynced(true))).toMatchSnapshot();
  });

  it('snapshot redux reducer multiple action test', () => {
    let state = appReducer(undefined, setAppSynced(true));
    expect(state).toMatchSnapshot();
    state = appReducer(state, setOfflineMode(true));
    expect(state).toMatchSnapshot();
  });
});
