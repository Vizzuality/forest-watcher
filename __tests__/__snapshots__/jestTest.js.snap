// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Redux Actions - Unit Tests snapshot redux action test 1`] = `
Object {
  "payload": true,
  "type": "app/SET_OFFLINE_MODE",
}
`;

exports[`Redux Actions - Unit Tests snapshot redux action test 2`] = `
Object {
  "payload": false,
  "type": "app/SET_OFFLINE_MODE",
}
`;

exports[`Redux Actions - Unit Tests snapshot redux action thunk test 1`] = `[Function]`;

exports[`Redux Reducers - Unit Tests snapshot redux reducer action test 1`] = `
Object {
  "actions": Array [],
  "coordinatesFormat": "decimal",
  "language": null,
  "offlineMode": false,
  "pristineCacheTooltip": true,
  "synced": true,
  "version": "1.6.3",
}
`;

exports[`Redux Reducers - Unit Tests snapshot redux reducer initial state test 1`] = `
Object {
  "actions": Array [],
  "coordinatesFormat": "decimal",
  "language": null,
  "offlineMode": false,
  "pristineCacheTooltip": true,
  "synced": false,
  "version": "1.6.3",
}
`;

exports[`Redux Reducers - Unit Tests snapshot redux reducer multiple action test 1`] = `
Object {
  "actions": Array [],
  "coordinatesFormat": "decimal",
  "language": null,
  "offlineMode": false,
  "pristineCacheTooltip": true,
  "synced": true,
  "version": "1.6.3",
}
`;

exports[`Redux Reducers - Unit Tests snapshot redux reducer multiple action test 2`] = `
Object {
  "actions": Array [],
  "coordinatesFormat": "decimal",
  "language": null,
  "offlineMode": true,
  "pristineCacheTooltip": true,
  "synced": true,
  "version": "1.6.3",
}
`;

exports[`Unit Tests snapshot runs 1`] = `
Object {
  "hello": true,
  "number": 232,
}
`;
